api/query_test.go

```diff
@@ -555,18 +555,16 @@ func MakeFileConnection() vds.ConnectionMaker {
 
 func setupTestServer(r *gin.Engine) {
 	r.Use(gin.Logger())
-	r.Use(ErrorHandler)
+	endpoint := Endpoint{MakeVdsConnection: MakeFileConnection()}
 
-	endpoint := Endpoint{ MakeVdsConnection: MakeFileConnection() }
+	r.POST("/slice", ErrorHandler, endpoint.SlicePost)
+	r.GET("/slice", ErrorHandler, endpoint.SliceGet)
 
-	r.POST("/slice", endpoint.SlicePost)
-	r.GET("/slice", endpoint.SliceGet)
+	r.POST("/fence", ErrorHandler, endpoint.FencePost)
+	r.GET("/fence", ErrorHandler, endpoint.FenceGet)
 
-	r.POST("/fence", endpoint.FencePost)
-	r.GET("/fence", endpoint.FenceGet)
-
-	r.POST("/metadata", endpoint.MetadataPost)
-	r.GET("/metadata", endpoint.MetadataGet)
+	r.POST("/metadata", ErrorHandler, endpoint.MetadataPost)
+	r.GET("/metadata", ErrorHandler, endpoint.MetadataGet)
```

Comment:
Maybe we should make a group and apply the error handler to that group? https://github.com/gin-gonic/gin#grouping-routes

Reply:
Thought to do so originally, but as far as I can tell, the group must have a name which is then a part of how you access the endpoints. So if I do group `e.Group("/stuff")`, I have to access all the endpoints added to it as `localhost:8080/stuff/fence`
Decided that we probably don't want that.
Now I see it [should be possible](https://github.com/go-ggz/ggz/blob/46af19ffdbe9b3d5316cfd4131eb3e0849297f17/pkg/router/routes/routes.go#L126-L133) to create 3 groups - one for metadata, one for fence, one for slice, and just combine post and get requests in them. Would you prefer that?

Reply:
You're right, we don't want that :D